program/rust/src/lib.rs

```diff
@@ -33,11 +36,34 @@ pub extern "C" fn c_entrypoint(input: *mut u8) -> u64 {
 
 #[no_mangle]
 pub extern "C" fn entrypoint(input: *mut u8) -> u64 {
-    let c_ret_val = unsafe { c_entrypoint(input) };
-    if c_ret_val == c_oracle_header::SUCCESSFULLY_UPDATED_AGGREGATE {
-        //0 is the SUCCESS value for solana
-        return 0;
-    } else {
-        return c_ret_val;
+    let (program_id, accounts, instruction_data) = unsafe { deserialize(input) };
+    let cmd_hdr_size = ::std::mem::size_of::<c_oracle_header::cmd_hdr>();
+    if instruction_data.len() < cmd_hdr_size {
+        panic!("insufficient data, could not parse instruction");
+    }
+
+    let cmd_data =
+        c_oracle_header::cmd_hdr::try_from_slice(&instruction_data[..cmd_hdr_size]).unwrap();
+
+    if cmd_data.ver_ != c_oracle_header::PC_VERSION {
+        //FIXME: I am not sure what's best to do here (this is copied from C)
```

Comment:
We need to discuss what is the best way to handle the version field.

Reply:
We agreed on slack that this will be handled in a future PR